[UI] Add volume tab in node page

ChengYanJin opened the pull request:
**Component**: ui, nodes

**Context**:  Add the volume tab in the Node Page.

**Summary**:
- Add some props in <VolumeListTable> component to make it reusable. 
- Add `no volume` for the <VolumeListTable>

**Acceptance criteria**: 
when there is no volume in the table:
![image](https://user-images.githubusercontent.com/18453133/94258510-408fd280-ff2d-11ea-93d2-0847bfb65e1a.png)

![image](https://user-images.githubusercontent.com/18453133/94258490-38379780-ff2d-11ea-8eb8-e70f7c780fab.png)
When clicking on the `create a new volume` button, the node field in the form should prefill with the current node.


Update after discuss with @Cuervino 
![image](https://user-images.githubusercontent.com/18453133/94692160-820bed80-0332-11eb-8bfe-fd99c4e4bbe8.png)

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #2777

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui/nodes/metrics: Add timespan selection for the graphs

Refs: #2776

ChengYanJin committed:
ui/nodes: Add the selected timespan in the URL

- When switch bwtween the nodes, keep the same tab, and same timespan if metric tab is selected
- When switch bwtween the nodes, if there no tab selected, by default `health` tab is selected

Refs: #2776

ChengYanJin committed:
ui/nodes: Extract the common y-axis specification for vega-lite linechart

Refs: #2776

ChengYanJin committed:
ui/package: Update the core-ui version

Fix the bug of progressbar in core-ui

Refs: #2777

ChengYanJin committed:
ui/nodes: Add temporary else if to get the node name from URL

Todo: Need to work on the path before change the URL back to /volumes

Refs: #2777

ChengYanJin committed:
ui/nodes: Fill the volume tab in New Node Page

Refs: #2777

ChengYanJin committed:
ui/nodes: Adapt the volumelistTable to be used by Node Page

Refs: #2777

ChengYanJin committed:
ui/styles: Change the padding-top of the tab

Refs: #2777

ChengYanJin committed:
ui/nodes: Address the state when there is no volume in the table

Refs: #2777

ChengYanJin committed:
ui/nodes: Add isSearchBar in the volumelistTable

Refs: #2777

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @thomasdanan

* @alexandre-allard-scality

* @tcarmet

* @nootal

* @teddyandrieux

* @lucieleonard

* @gdemonet

* @sayf-eddine-scality

* @nicolast

* @jbertran

* @monpote

* @carlito-scality

* @slaperche-scality

* @cmonfort

MonPote commented on ui/src/ducks/app/monitoring.js:
```diff
@@ -722,6 +740,33 @@ export function* fetchNodeStats({ payload }) {
+      },
+    });
+  }
+  yield delay(REFRESH_METRCIS_GRAPH);
```
typo : `REFRESH_METRCIS_GRAPH` -> `REFRESH_METRICS_GRAPH`

ChengYanJin replied:
I fixed this typo in #2809 since it's not related to this PR.

MonPote commented on ui/src/components/VolumeListTable.js:
```diff
@@ -256,6 +256,26 @@ function Table({ columns, data, nodeName, rowClicked, volumeName, theme }) {
+              <td
+                style={{
+                  textAlign: 'center',
+                  background: theme.brand.primaryDark2,
```
Do we want to use the `primaryDark2` or to use the same background color as the Table (`primary`)

ChengYanJin replied:
After talking with Valamir, since no volume in the node is not an error state. I removed the Exclamation icon and change the background color to primaryDark1 which the color of table.

ChengYanJin committed:
ui/nodes: Remove the Exclamation icon when there is no volume in the node

- Change the background to the same as table

Refs: #2777

bert-e commented:
# Branches have diverged



This pull request's source branch `improvement/2777-add-volume-tab-in-node-page` has diverged from
`development/2.6` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.6` into `improvement/2777-add-volume-tab-in-node-page`
* Rebase `improvement/2777-add-volume-tab-in-node-page` onto `origin/development/2.6`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

ChengYanJin committed:
ui/theme: Update the them for light/dark mode

- Update the dark mode with final version
- Update the status color of light mode with proposal, but keep the primary/text/border as the previous version,
  since it doesn't work well.

thomasdanan reviewed (dismissed):
just perfect !!!
We still have this little bug about the capacity bar

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/2777-add-volume-tab-in-node-page` into
`development/2.6`
* [w/2.7/improvement/2777-add-volume-tab-in-node-page](https://github.com/scality/metalk8s/compare/development/2.7...w/2.7/improvement/2777-add-volume-tab-in-node-page) will be merged into `development/2.7` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @nicolast

* @lucieleonard

* @alexandre-allard-scality

* @monpote

* @slaperche-scality

* @teddyandrieux

* @cmonfort

* @thomasdanan

* @carlito-scality

* @jbertran

* @sayf-eddine-scality

* @gdemonet

* @nootal

gdemonet reviewed (commented):
It seems you lost some commits, I see nothing about volume tab in the diff... :confused:

ChengYanJin committed:
fix the bug of URL

bert-e commented:
# History mismatch



Merge commit #133b3959bdf868772754338cedbfdd71d4a4a646 on the integration branch
`w/2.7/improvement/2777-add-volume-tab-in-node-page` is merging a branch which is neither the current
branch `improvement/2777-add-volume-tab-in-node-page` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/2777-add-volume-tab-in-node-page` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #887ee570b36c97bbba3c781720587febbf739d97 on the integration branch
`w/2.7/improvement/2777-add-volume-tab-in-node-page` is merging a branch which is neither the current
branch `improvement/2777-add-volume-tab-in-node-page` nor the development branch
`development/2.7`.

It is likely due to a rebase of the branch `improvement/2777-add-volume-tab-in-node-page` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

ChengYanJin commented:
> just perfect !!!
> We still have this little bug about the capacity bar

Hi @thomasdanan,
The capacity bar is fixed, just you'll need to update your local dependencies with:
- rm -rf node_modules
- npm install

ChengYanJin commented:
> It seems you lost some commits, I see nothing about volume tab in the diff... 😕

Hi @gdemonet, 

Indeed, I lost my commits before. Now we get them!
